### PR TITLE
Add mapper functionality to the collection dashboard

### PR DESCRIPTION
**Pull request by nvisal1**
***Purpose***
This PR adds the option for admins to assign Mappers in the Collection Dashboard

***Story***
This PR addresses Clubhouse story https://app.clubhouse.io/clarkcan/story/1765/add-new-access-group-mapper-to-the-system

***Screenshots***
Screen Shot 2020-02-25 at 3 25 05 PM
Screen Shot 2020-02-25 at 3 25 12 PM

### CODE REVIEW

File: src/app/admin/components/user-privileges/user-privileges.component.html [hunk 1]
```diff
@@ -59,6 +67,17 @@
       <clark-collections-grid [preselect]="collectionsWithSelectedRole" preselectReason="This user is already a {{ selectedRole }} in this collection." (selected)="selectCollection($event)"></clark-collections-grid>
     </ng-template>
 
+    <ng-template>
+      <div class='mapper-confirmation'>
+        <h2 class='mapper-confirmation__header'>Are you sure?</h2>
+        <p class='mapper-confirmation__sub'>You are about to promote this user to a Mapper.</p>
+        <div class='btn-group center'>
+          <button class='button neutral'>Cancel</button>
```
Comment (pzalep1) [inline src/app/admin/components/user-privileges/user-privileges.component.html]: Aria-label please

File: src/app/admin/components/user-privileges/user-privileges.component.html [hunk 2]
```diff
@@ -59,6 +67,17 @@
...
+          <button class='button good'>Submit</button>
```
Comment (pzalep1) [inline src/app/admin/components/user-privileges/user-privileges.component.html]: Aria-label please

File: src/app/admin/components/user-privileges/user-privileges.component.html [hunk 3]
```diff
@@ -73,6 +92,10 @@
       <i class="far fa-user-crown"></i>
       Curator
     </div>
+    <div [ngClass]="{'role-selector__role--active': selectedRole === 'mapper'}" (activate)="selectedRole = 'mapper'; advance(2)" class="role-selector__role">
```
Comment (pzalep1) [inline src/app/admin/components/user-privileges/user-privileges.component.html]: This should be a button.

File: src/app/components/navbar/navbar.component.html [hunk 4]
```diff
@@ -21,27 +21,20 @@
 
       <ng-template #userTemplate>
         <ng-container *ngIf="loggedin">
-          <button class="contributor" id="contributor-link" #contributorDropdownButtonElement (activate)="contributorDropdown(true)"><span>Contribute</span> <i class="fas fa-chevron-down"></i></button>
-          <a class="right library" id="library-link" routerLink="/library"></a>
           <button id="clark-user-dropdown" #userDropdownButtonElement class="user" (activate)="userDropdown(true)">
```
Comment (pzalep1) [inline src/app/components/navbar/navbar.component.html]: I know you didn't write this, but aria label please.

File: src/app/components/navbar/navbar.component.html [hunk 5]
```diff
@@ -21,27 +21,20 @@
...
-              </ul>
-            </div>
-          </clark-context-menu>
           
           <clark-context-menu *ngIf="userDown" [anchor]="userDropdownButtonElement" [offset]="{ top: 10, left: 5 }" (close)="userDropdown()">
             <div #contextMenu id="context-popup">
               <ul>
                 <li tabindex="0" *ngIf="authService.hasCuratorAccess()" routerLink="/admin" (activate)="userDropdown()"><i class="far fa-tools"></i> {{ authService.hasEditorAccess() ? 'Admin' : 'Curator' }} Dashboard</li>
-                <li tabindex="0" [routerLink]="['users', authService.user?.username]" (activate)="userDropdown()"><i class="far fa-user"></i>Your Profile</li>
+                <li tabindex="0" routerLink="/library" (activate)="userDropdown()"><i class="fas fa-books"></i>Library<span *ngIf="notifications?.length" class="notifications"> {{ libraryNotificationCount }}</span></li>
```
Comment (pzalep1) [inline src/app/components/navbar/navbar.component.html]: I think this is new nav bar stuff

File: src/app/admin/components/user-privileges/user-privileges.component.html [hunk 6]
```diff
@@ -59,20 +67,34 @@
...
+    </ng-template>
+
   </clark-carousel>
 
 </div>
 
 <ng-template #roleSelector>
   <div class="role-selector">
-    <div [ngClass]="{'role-selector__role--active': selectedRole === 'reviewer'}" (activate)="selectedRole = 'reviewer'; advance()" class="role-selector__role">
+    <button [ngClass]="{'role-selector__role--active': selectedRole === 'reviewer'}" (activate)="selectedRole = 'reviewer'; advance()" class="role-selector__role">
```
Comment (pzalep1) [inline src/app/admin/components/user-privileges/user-privileges.component.html]: We'll need an aria label on this as well

File: src/app/admin/components/user-privileges/user-privileges.component.scss [hunk 7]
```diff
@@ -143,4 +143,17 @@
...
     margin-bottom: 20px;
   }
+}
+
+.mapper-confirmation {
+  text-align: center;
+  color: $dark-grey;
+}
+
+.mapper-confirmation__header {
```
Comment (EVogel1999) [inline src/app/admin/components/user-privileges/user-privileges.component.scss]: Empty CSS class?

File: src/app/admin/components/user-privileges/user-privileges.component.html [hunk 8]
```diff
@@ -38,6 +38,14 @@
           </div>
         </div>
       </ng-template>
+
+      <ng-template>
+        <div class="privileges__top-inner">
+          <div (activate)="regress(2)" class="privileges__subtitle privileges__subtitle--interactable">
```
Comment (pzalep1) [inline src/app/admin/components/user-privileges/user-privileges.component.html]: This should be a button.

File: src/app/admin/components/user-privileges/user-privileges.component.html [hunk 9]
```diff
@@ -22,9 +22,9 @@
 
       <ng-template>
         <div class="privileges__top-inner">
-          <div (activate)="regress()" class="privileges__subtitle privileges__subtitle--interactable">
+          <button (activate)="regress()" class="privileges__subtitle privileges__subtitle--interactable">
```
Comment (pzalep1) [inline src/app/admin/components/user-privileges/user-privileges.component.html]: aria-label please

File: src/app/admin/components/user-privileges/user-privileges.component.html [hunk 10]
```diff
@@ -38,6 +38,14 @@
           </div>
         </div>
       </ng-template>
+
+      <ng-template>
+        <div class="privileges__top-inner">
+          <button (activate)="regress(2)" class="privileges__subtitle privileges__subtitle--interactable">
```
Comment (pzalep1) [inline src/app/admin/components/user-privileges/user-privileges.component.html]: aria-label please

File: src/app/admin/components/user-privileges/user-privileges.component.html [hunk 11]
```diff
@@ -59,20 +67,34 @@
       <clark-collections-grid [preselect]="collectionsWithSelectedRole" preselectReason="This user is already a {{ selectedRole }} in this collection." (selected)="selectCollection($event)"></clark-collections-grid>
     </ng-template>
 
+    <ng-template>
```
Comment (EVogel1999) [inline src/app/admin/components/user-privileges/user-privileges.component.html]: Does this template need an id?

File: src/app/admin/components/user-privileges/user-privileges.component.html [hunk 12]
```diff
@@ -38,6 +38,14 @@
           </div>
         </div>
       </ng-template>
+
+      <ng-template>
```
Comment (EVogel1999) [inline src/app/admin/components/user-privileges/user-privileges.component.html]: Same with this template.

**Pull request merged**